spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteRequirements.java

```diff
@@ -47,4 +51,8 @@ public SortOrder[] ordering() {
   public boolean hasOrdering() {
     return ordering.length != 0;
   }
+
+  public long advisoryPartitionSize() {
+    return distribution instanceof UnspecifiedDistribution ? 0 : advisoryPartitionSize;
```

Comment:
Spark will complain if we request a value > 0 with `UnspecifiedDistribution`.

Reply:
Good thing for a comment.

Reply:
Will add.